common/src/main/cpp/openxr_fb_passthrough_extension_wrapper.cpp

```diff
@@ -0,0 +1,287 @@
+#include <godot_cpp/classes/os.hpp>
+#include <godot_cpp/classes/scene_tree.hpp>
+#include <godot_cpp/classes/viewport.hpp>
+#include <godot_cpp/classes/xr_server.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+OpenXRFbPassthroughExtensionWrapper *OpenXRFbPassthroughExtensionWrapper::singleton = nullptr;
+
+OpenXRFbPassthroughExtensionWrapper *OpenXRFbPassthroughExtensionWrapper::get_singleton() {
+	if (singleton == nullptr) {
+		singleton = memnew(OpenXRFbPassthroughExtensionWrapper());
+	}
+	return singleton;
+}
+
+OpenXRFbPassthroughExtensionWrapper::OpenXRFbPassthroughExtensionWrapper() :
+		OpenXRExtensionWrapperExtension() {
+
+	ERR_FAIL_COND_MSG(singleton != nullptr, "An OpenXRFbPassthroughExtensionWrapper singleton already exists.");
+
+	request_extensions[XR_FB_PASSTHROUGH_EXTENSION_NAME] = &fb_passthrough_ext;
+	request_extensions[XR_FB_TRIANGLE_MESH_EXTENSION_NAME] = &fb_triangle_mesh_ext;
+
+	singleton = this;
+}
+
+OpenXRFbPassthroughExtensionWrapper::~OpenXRFbPassthroughExtensionWrapper() {
+	cleanup();
+}
+
+void OpenXRFbPassthroughExtensionWrapper::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("is_passthrough_supported"), &OpenXRFbPassthroughExtensionWrapper::is_passthrough_supported);
+	ClassDB::bind_method(D_METHOD("is_passthrough_enabled"), &OpenXRFbPassthroughExtensionWrapper::is_passthrough_enabled);
+}
+
+godot::Dictionary OpenXRFbPassthroughExtensionWrapper::_get_requested_extensions() {
+	godot::Dictionary result;
+	for (auto ext: request_extensions) {
+		godot::String key = ext.first;
+		uint64_t value = reinterpret_cast<uint64_t>(ext.second);
+		result[key] = (godot::Variant)value;
+	}
+	return result;
+}
+
+void OpenXRFbPassthroughExtensionWrapper::cleanup() {
+	fb_passthrough_ext = false;
+	fb_triangle_mesh_ext = false;
+}
+
+void OpenXRFbPassthroughExtensionWrapper::_on_instance_created(uint64_t p_instance) {
+	XrInstance instance = (XrInstance)p_instance;
+	if (fb_passthrough_ext) {
+		bool result = initialize_fb_passthrough_extension(instance);
+		if (!result) {
+			UtilityFunctions::print("Failed to initialize fb_passthrough extension");
+			fb_passthrough_ext = false;
+		}
+	}
+
+	if (fb_triangle_mesh_ext) {
+		bool result = initialize_fb_triangle_mesh_extension(instance);
+		if (!result) {
+			UtilityFunctions::print("Failed to initialize fb_triangle_mesh extension");
+			fb_triangle_mesh_ext = false;
+		}
+	}
+}
+
+void OpenXRFbPassthroughExtensionWrapper::_on_session_created(uint64_t p_session) {
+	XrSession session = (XrSession)p_session;
+	if (fb_passthrough_ext) {
+		// If some other extension is already emulating alpha blend mode, we won't.
+		if (get_openxr_api()->is_emulating_environment_blend_mode_alpha_blend()) {
+			return;
+		}
+
+		// If the runtime already supports the alpha blend mode, we won't emulate it.
+		if (get_openxr_api()->is_real_environment_blend_mode_alpha_blend_supported()) {
+			return;
+		}
```

Comment:
See my latest comments in https://github.com/godotengine/godot/pull/87630 to consolidate this logic.

Reply:
This is consolidated in my lastest push per your suggestion!